### PR TITLE
[improve](json)improve json support empty keys

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
support empty key for json 
behavior reference MSYQL 
 
Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/util/jsonb_utils.h [hunk 1]
```diff
@@ -191,7 +191,11 @@ class JsonbToJson {
             if (iter->klen()) {
                 string_to_json(iter->getKeyStr(), iter->klen());
             } else {
-                os_.write(iter->getKeyId());
+                if (iter->getKeyId() == 0) {
```
Comment (eldenmoon) [inline be/src/util/jsonb_utils.h:194]: is key == 0 special ? if special add comment to explain

Comment (amorynan) [reply be/src/util/jsonb_utils.h:194]: Yes, we do not use keyid before , just use keystr for jsonkey, and se I make keyid == 0 is special for empty key

Comment (eldenmoon) [reply be/src/util/jsonb_utils.h:194]: maybe use sMaxKeyId for safe and readable, 0 is not too special

File: be/src/util/jsonb_writer.h [hunk 2]
```diff
@@ -88,8 +88,14 @@ class JsonbWriterT {
             uint32_t size = sizeof(uint8_t);
             if (key_id < 0) {
                 os_->put(len);
-                os_->write(key, len);
-                size += len;
+                if (len == 0) {
+                    JsonbKeyValue::keyid_type idx = 0;
```
Comment (eldenmoon) [inline be/src/util/jsonb_writer.h:92]: use sMaxKeyId

Comment (eldenmoon) [reply be/src/util/jsonb_writer.h:92]: add comment

File: be/src/util/jsonb_utils.h (hunk 1 above)
Comment (amorynan) [reply be/src/util/jsonb_utils.h:194]: OK I will try

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

File: be/src/util/jsonb_writer.h [hunk 3]
```diff
@@ -88,9 +88,16 @@ class JsonbWriterT {
             uint32_t size = sizeof(uint8_t);
             if (key_id < 0) {
                 os_->put(len);
-                os_->write(key, len);
-                size += len;
-            } else if (key_id <= JsonbKeyValue::sMaxKeyId) {
+                if (len == 0) {
+                    // NOTE: we use sMaxKeyId to represent an empty key
+                    JsonbKeyValue::keyid_type idx = JsonbKeyValue::sMaxKeyId;
```
Comment (eldenmoon) [inline be/src/util/jsonb_writer.h:93]: add comment to where sMaxKeyId defined

Comment (amorynan) [reply be/src/util/jsonb_writer.h:93]: done!

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**